qa/workunits/ceph-helpers.sh

```diff
@@ -1452,13 +1472,13 @@ function main() {
 
     local code
     if run $dir "$@" ; then
-        code=0
+        teardown $dir || return 1
+        return 0
     else
         display_logs $dir
-        code=1
+        teardown_error $dir
+        return 1
     fi
-    teardown $dir || return 1
```

Comment:
@tchaikov 
It is not my intention to change any of the behaviour of the tests.
Other than:
  - Fix FreeBSD issues
  - Keep data around when something fails, so one can look at the logfiles
So if it does change standard behaviour on Linux, when tests go as planned, that should be a mistake.

But I'm fine with taking this change to a next set of commits for the next pull.
Note that I talked about it with a few people at Ceph day, and it would be nice if tests also start taking care of situations where things actually fail, and people doing testing/debugging/porting would have easy access to more information.